api_client/python/timesketch_api_client/sigma.py

```diff
@@ -95,3 +98,41 @@ def file_relpath(self):
             return ''
 
         return sigma_data.get('file_relpath', '')
+
+    def from_rule(self, rule_uuid):
+        """Get a Sigma object from a rule uuid.
+
+        Args:
+            rule_uuid: Id of the sigma rule.
+
+        """
+        # TODO: Write docstring
+        self.rule_uuid = rule_uuid
+        # TODO: not sure which one is the better one
+        self._resource_uri = f'sigma/{rule_uuid}'
+        self.resource_uri = f'sigma/{rule_uuid}'
+        super().__init__(
+            api=self.api, resource_uri=self._resource_uri)
+
+        self.lazyload_data()
+
+    def from_text(self, rule_text):
+        """Get a Sigma object from a rule text.
+
+        Args:
+            rule_text: Rule text to be parsed.
+
+        """
+        resource_url = '{0:s}/sigma_by_text/'.format(self.api.api_root)
+        data = {'title': 'Get_Sigma_by_text', 'content': rule_text}
+        response = self.api.session.post(resource_url, data=data)
+        print(response.__dict__)
+        response_dict = error.get_response_json(response, logger)
+        #timeline_dict = response_dict['objects'][0]
+        #timeline_obj = sigma.Sigma(
+        #    timeline_id=timeline_dict['id'],
```

Comment:
not sure why you do a ` timeline_dict`, since this is not a timeline... 

but here you should do something like

```
objects = response_dict.get('objects')

if not objects:
  # do something
```

and then depending on how the API works, you may only be interested in the first object, or all of the returned ones.

Then the `timeline_obj` should definitely not be another sigma object, you already have that... now you just assign these same variables that you set in `___init__` to the results from here.

Reply:
I think that should now be done